arrow/src/main/java/org/apache/iceberg/arrow/vectorized/parquet/VectorizedPageIterator.java

```diff
@@ -202,7 +209,7 @@ public int nextBatchFloats(
     if (actualBatchSize <= 0) {
       return 0;
     }
-    if (eagerDecodeDictionary) {
+    if (dictionaryDecodeMode == DictionaryDecodeMode.EAGER) {
```

Comment:
Coverage shows that this branch isn't taken -- line 213 is never used by the existing tests. The equivalent line for doubles is taken, though. Can you find out what's happening?

Reply:
Actually, this was in my local copy. When I reverted the changes I see that the tests are hitting this line. But the decimal line below is not showing up in coverage.